snapcraft/linters/base.py

```diff
@@ -103,14 +104,44 @@ def run(self) -> List[LinterIssue]:
         :return: A list of linter issues flagged by this linter.
         """
 
-    def _is_file_ignored(self, elf_file: ElfFile) -> bool:
-        """Check if the file name matches an ignored file pattern."""
-        for pattern in self._lint.ignored_files(self._name):
-            if fnmatch.fnmatch(str(elf_file.path), pattern):
+    def _is_file_ignored(
+        self, filepath: Union[ElfFile, Path], category: str = ""
+    ) -> bool:
+        """Check if the file name matches an ignored file pattern.
+
+        :param category:
+            For linters that implement categories, the specific category that
+            should be checked for the ignored patterns. `filepath` will be checked
+            against the linter's "main" patterns (based on the linter's name) *and*
+            against the category-specific patterns.
+        """
+        ignored_files = self._lint.ignored_files(self._name)
+
+        if category:
+            # No "extend()" because we don't want to affect the original list.
+            ignored_files = ignored_files + self._lint.ignored_files(category)
+
+        if isinstance(filepath, ElfFile):
+            path = filepath.path
+        else:
+            path = filepath
```

Comment:
I wonder if we shouldn't make `filepath` just a `Path`, and change the caller to use `elf_file.path` when necessary.

Reply:
I thought about doing this, but I didn't want to start changing a bunch of files (noisy on the review). Do you think we should just pull the trigger and do this now?